test/Chainweb/Test/Pact/RemotePactTest.hs

```diff
@@ -263,24 +259,52 @@ pollWithRetry
     :: PactTestApiCmds
     -> ClientEnv
     -> RequestKeys
-    -> IO (Either ClientError PollResponses)
+    -> IO PollResponses
 pollWithRetry cmds env rks = do
   sleep 3
   go maxPollRetries
     where
-      go retries = do
-          result <- runClientM (pollApiCmd cmds (Poll (_rkRequestKeys rks))) env
+      go !retries = do
+          let retry failure = do
+                if retries == 0
+                  then do
+                    putStrLn $ "poll failing after " ++ show maxSendRetries
+                               ++ " retries"
+                    fail failure
+                  else sleep 1 >> go (retries - 1)
+          let rkks = _rkRequestKeys rks
+          result <- runClientM (pollApiCmd cmds (Poll rkks)) env
           case result of
-              Left _ ->
-                  if retries == 0 then do
-                      putStrLn $ "poll failing after " ++ show maxSendRetries ++ " retries"
-                      return result
-                  else do
-                      sleep 1
-                      go (retries - 1)
-              Right _ -> do
-                  putStrLn $ "poll succeeded after " ++ show (maxSendRetries - retries) ++ " retries"
-                  return result
+              Left e -> retry (show e)
+              Right (PollResponses mp) -> do
+                  ok <- checkResults (toList rkks) mp
+                  if ok
+                    then do
+                        putStrLn $ concat
+                            [ "poll succeeded after "
+                            , show (maxSendRetries - retries)
+                            , " retries" ]
+                        return $ PollResponses mp
+                    else do
+                        putStrLn "poll check failed, continuing"
+                        let msg = concat [ "poll check failure. keys: "
+                                         , show rks
+                                         , "\nresult map was: "
+                                         , show mp
+                                         ]
+                        retry msg
+
+      checkCommandResult x cr =
+          _crReqKey cr == x && isOkPactResult (_crResult cr)
```

Comment:
the req key should be it's own check with informative error if it fails

Reply:
We dump the inputs if it fails so a debug session should be possible here. Again, this is test code, I'm loath to poke it much further. Emily wants to come in and rewrite this module, it's not a bad idea